lib/config/configuration.go

```diff
@@ -938,6 +1040,49 @@ func Configure(clf *CommandLineFlags, cfg *service.Config) error {
 		}
 	}
 
+	// If application name was specified on command line, add to configuration.
+	if clf.AppName != "" {
+		cfg.Apps.Enabled = true
+
+		// Check if the application name contains spaces. Don't allow spaces because
+		// for trusted clusters the name is used to construct the vanity domain that
+		// the application will be running under.
+		if ok := strings.Contains(clf.AppName, " "); ok {
```

Comment:
Just as a little simplification suggestion, instead of repeating all the same logic for checking app name, etc., you could just set all these cli flag values in `fileConf` and then `ApplyFileConfig(fileConf, cfg)` would do the rest for you and reuse all that code you wrote above and this whole block will be gone.

```go
if clf.AppName != "" {
  fileConf.Apps = Apps{
    Service: Service{EnabledFlag: "yes"},
    Apps: []*App{
      Name: clf.AppName,
      ...
    },
  }
}
```